packages/app-mobile/root.tsx

```diff
@@ -810,6 +814,13 @@ class AppComponent extends React.Component {
 		return false;
 	}
 
+	private onItemChange_({ itemType }: {itemType: ModelType}) {
+		if (itemType !== ModelType.Note) {
+			return;
+		}
+		updateRecentsWidgetWithDebounce();
```

Comment:
Is there any way to detect if the widget is installed, and if it's not, to skip this call?

Reply:
Seems the only way is to [store the widget status in shared preferences](https://stackoverflow.com/questions/8836724/check-widget-is-placed-on-android-screen) and use that. However, I don't think this is a good idea:

- It complicates the update flow, as the widget would have no data when first added, so we'd need to expose some way for the widget to trigger these queries and I'm not sure how to cleanly do that.
- I'm not sure if we're saving much, if anything, since React would still need to use native code to read/write shared preferences just to check if a widget is active.